homeassistant/components/number/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""Component to allow numeric input for platforms."""
+from datetime import timedelta
+import logging
+from typing import Any, Dict
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    ATTR_MAX,
+    ATTR_MIN,
+    ATTR_STEP,
+    ATTR_VALUE,
+    DEFAULT_MAX_VALUE,
+    DEFAULT_MIN_VALUE,
+    DEFAULT_STEP,
+    DOMAIN,
+    SERVICE_SET_VALUE,
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Number entities."""
+    component = hass.data[DOMAIN] = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+    await component.async_setup(config)
+
+    component.async_register_entity_service(
+        SERVICE_SET_VALUE,
+        {vol.Required(ATTR_VALUE): vol.Coerce(float)},
+        "async_set_value",
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    return await hass.data[DOMAIN].async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class NumberEntity(Entity):
+    """Representation of a Number entity."""
+
+    @property
+    def capability_attributes(self) -> Dict[str, Any]:
+        """Return capability attributes."""
+        data = {
+            ATTR_MIN: self.min_value,
+            ATTR_MAX: self.max_value,
+            ATTR_STEP: self.step,
+        }
+        return data
+
+    @property
+    def min_value(self) -> float:
+        """Return the minimum value."""
+        return DEFAULT_MIN_VALUE
+
+    @property
+    def max_value(self) -> float:
+        """Return the maximum value."""
+        return DEFAULT_MAX_VALUE
```

Comment:
I'm currently a bit thorn about have a required min/max value. Initially, it sounded like a good thing to me, however, when considering default values I started to doubt that.

What would be the reason to require it? We don't require it on `input_number`?

I know this is not much of a review comment, as this was raised/suggested in the architectural issue by @balloob.

Reply:
To be honest, I don't quite understand your suggestion. Do you mean that the `number` should have no limits?
Yes, `input_number` requires min and max. And any property I could think of that we could control with the `number` have natural bounds.

Reply:
Discussed it with balloob in chat as well, let's leave it like this for now.